05-lifecycle_of_a_Patch.Rmd

```diff
@@ -0,0 +1,103 @@
+
+### How to review a patch?
+
+One of the bottlenecks in the R development process is the lack of code reviews. If you browse Bugzilla, you will see that numerous issues have a fix, but cannot be merged into the main source code repository, because no one has reviewed the proposed solution. Reviewing a patch can be just as informative as providing a patch and it will allow you to give constructive comments on another developer’s work. This guide provides a checklist for submitting a patch review. It is a common misconception that in order to be useful, a patch review has to be perfect. This is not the case at all! It is helpful to just test the patch and/or play around with the code and leave comments in the patch or on Bugzilla.
+
+If a bug report or an issue has a patch attached that has not been reviewed, you can help by making sure that the patch:
+
+1. follows the style guides.
+
+2. is a good solution to the problem it is trying to solve.
+
+3. includes proper tests.
+
+4. includes proper documentation changes.
+
+5. is reproducible on following the stated steps. If not then, you may comment which step is not reproducible on the version of R that you are using.
+
+Also refer to [Making good patches](#GoodPatches) for more ideas. Doing all of this allows the core developer(s) and/ or maintainer(s) to more quickly look for subtle issues that only people with extensive experience working on R's codebase will notice.
+
+## Leaving a patch review on Bugzilla
+
+When you review a patch, you should provide additional details and context of your review process and leave comments. For example:
+
+1. If you tested the patch, report the result and the system and version tested on, such as 'Windows 10', 'Ubuntu 16.4', or 'Mac High Sierra'.
+
+2. If you request changes, try to suggest how.
+
+3. Comment on what is 'good' about the patch, not just the 'bad'. Doing so will make it easier for the patch author to find the good in your comments.
+
+## Dismissing review from another core developer
+
+A core developer can dismiss another core developer’s review if they confirmed that the requested changes have been made. When a core developer has assigned the patch to themselves, then it is a sign that they are actively looking after the patch, and their review should not be dismissed.
+
+## Acceptance or rejection of your patch
+
+Once your patch has reached an acceptable state, it will either be applied or rejected. If it is rejected, please do not take it personally. Your work is still appreciated regardless of whether your patch is applied. Balancing what does and does not go into R is tricky and everyone's contributions cannot always be accepted.
+
+But if your patch is accepted and applied it will then go on to be released with the next patched release and eventually the next major release of R. It may also be backported to older versions of R as a bugfix if the core developer doing the patch acceptance believes it is warranted.
+

```

Comment:
Perhaps would add a sentences or two about how long it can take to be included or rejected. There isn't any hard data yet, but mentioning that even if the bug report takes years (months?) till a change happens it is appreciated.